torch/_inductor/codegen/pallas.py

```diff
@@ -529,16 +530,65 @@ def codegen_kernel(self, name: Optional[str] = None) -> str:  # type: ignore[ove
                 code.writeline(str(line))
 
         jit_wrapper_name = f"{kernel_name}_jit_wrapper"
-        code.writeline("@functools.partial(jax.jit, static_argnums=(0, 1))")
-        code.writeline(f"def {jit_wrapper_name}(out_shape, out_dtype, *kernel_refs):")
+        code.writeline("@functools.partial(jax.jit, static_argnums=(0, 1, 2))")
+        code.writeline(
+            f"def {jit_wrapper_name}(out_shape, out_dtype, device_type, *kernel_refs):"
+        )
         with code.indent():
-            code.writeline("out_spec = jax.ShapeDtypeStruct(out_shape, out_dtype)")
-            code.writeline("return pl.pallas_call(")
+            code.writeline(
+                "# On GPU (Triton backend), Pallas requires each dimension to be power of 2"
+            )
+            code.writeline("if device_type == 'cuda':")
```

Comment:
We should know the device at compile time, no need for this to be a runtime check.